java-service.gitlab-ci.yml

```diff
@@ -42,7 +42,7 @@ dependency_scanning:
   stage: build
   allow_failure: true
   services:
-    - docker:stable-dind
+    - docker:18-dind
```

Comment:
I understand that the dependency scanning docs say to use this. I wonder if we actually need to use it, since we chose to use (at least on our machines) [exposure of the docker socket](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-docker-socket-binding) instead of [dind](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-docker-in-docker-executor)